### PR TITLE
feat(rust_extractor): remove dependency on rustc_driver to avoid dylink

### PR DESCRIPTION
**Pull request by wcalandro**
Remove dependency on a dynamic link to rustc_driver

### CODE REVIEW

**Comment by wcalandro** [comment]
Looks like still might have a dynamic link dependency on `libLLVM-12-rust-1.54.0-nightly.so`

File: kythe/rust/extractor/src/lib.rs [hunk 1]
```diff
@@ -29,75 +36,81 @@ use std::path::PathBuf;
...
+        return Err(anyhow!("The first argument must be an empty string"));
     }
 
-    let mut callback_shim = CallbackShim::new(output_dir);
+    // Generate the configuration to parse the options
+    let mut options = getopts::Options::new();
+    for option in config::rustc_optgroups() {
+        (option.apply)(&mut options);
+    }
+    let matches = options.parse(&rustc_arguments[1..]).context("Failed to parse options")?;
```
Comment (jaysachs) [inline kythe/rust/extractor/src/lib.rs:51]: Are we going to need this same or very similar logic for this creating of the `config` and running the compiler for the indexer as well?

Comment (wcalandro) [reply kythe/rust/extractor/src/lib.rs:51]: No, because the indexer does not run the compiler

File: kythe/rust/extractor/src/lib.rs [hunk 2]
```diff
@@ -29,75 +36,81 @@ use std::path::PathBuf;
 /// first element must be an empty string.
```
Comment (jaysachs) [inline kythe/rust/extractor/src/lib.rs:36]: Simply stating this doesn't feel like enough explanation for what seems like a strange arbitrary constraint. What's the "why" for this?

File: kythe/rust/extractor/src/lib.rs [hunk 3]
```diff
@@ -29,75 +36,81 @@ use std::path::PathBuf;
...
-        // falsely claims that there is a mismatch between rustc_save_analysis's
-        // `rls_data::config::Config` and ours, even though we use the same version.
-        // This forces us to use the environment variable method of configuration
-        // instead.
-        std::env::set_var(
-            "RUST_SAVE_ANALYSIS_CONFIG",
-            r#"{"output_file":null,"full_docs":true,"pub_only":false,"reachable_only":false,"distro_crate":false,"signatures":false,"borrow_data":false}"#,
-        );
+            std::env::set_var(
+                "RUST_SAVE_ANALYSIS_CONFIG",
```
Comment (jaysachs) [inline kythe/rust/extractor/src/lib.rs:93]: I'm not fully following the flow here, but this doesn't need to happen *before* `run_compiler` is invoked?

File: kythe/rust/extractor/src/lib.rs (hunk 2 above)
Comment (wcalandro) [reply kythe/rust/extractor/src/lib.rs:36]: The first argument always gets ignored, which is why it should be an empty string. But I agree, this comment should be better.

File: kythe/rust/extractor/src/lib.rs (hunk 3 above)
Comment (wcalandro) [reply kythe/rust/extractor/src/lib.rs:93]: This environment variable isn't read until we actually process the crate with the save analysis crate, so we set it right before we do the analysis.

**Review by jaysachs — APPROVED** [review]

**Comment by wcalandro** [comment]
This change is no longer the best way forward as it introduces other dynamically linked dependencies. Closing for now

**Pull request closed without merging**